src/coreclr/jit/llvmtypes.cpp

```diff
@@ -212,81 +212,59 @@ Type* Llvm::getLlvmTypeForVarType(var_types type)
         case TYP_LONG:
         case TYP_ULONG:
             return Type::getInt64Ty(_llvmContext);
-        case var_types::TYP_FLOAT:
+        case TYP_FLOAT:
             return Type::getFloatTy(_llvmContext);
-        case var_types::TYP_DOUBLE:
+        case TYP_DOUBLE:
             return Type::getDoubleTy(_llvmContext);
-        case TYP_BYREF:
         case TYP_REF:
+        case TYP_BYREF:
             return Type::getInt8PtrTy(_llvmContext);
-        case TYP_VOID:
-            return Type::getVoidTy(_llvmContext);
-        default:
+        case TYP_BLK:
+        case TYP_STRUCT:
             failFunctionCompilation();
+        default:
+            unreached();
     }
 }
 
-Type* Llvm::getLlvmTypeForLclVar(GenTreeLclVar* lclVar)
+Type* Llvm::getLlvmTypeForLclVar(LclVarDsc* varDsc)
 {
-    var_types nodeType = lclVar->TypeGet();
-
-    if (nodeType == TYP_STRUCT)
+    if (varDsc->TypeGet() == TYP_STRUCT)
+    {
+        return getLlvmTypeForStruct(varDsc->GetLayout());
+    }
+    // TODO-LLVM: enable. Currently broken because RyuJit inserts RPI helpers for RPI methods,
```

Comment:
Presumably this is a very easy fix (do not call `GenerateRuntimeExportThunk` from `CompileSingleMethod`), but I did not check.